src/services/CapabilitiesManager.ts

```diff
@@ -73,9 +129,10 @@ export async function setRemoteCapabilities(joinRoomResponse: JoinRoomFullRespon
 		return
 	}
 
-	remoteCapabilities[token] = { spreed: response.data.ocs.data }
-	remoteCapabilities[token].hash = joinRoomResponse.headers['x-nextcloud-talk-proxy-hash']
+	remoteCapabilities[remoteServer] = { spreed: response.data.ocs.data }
+	remoteCapabilities[remoteServer].hash = joinRoomResponse.headers['x-nextcloud-talk-proxy-hash']
```

Comment:
```suggestion
	remoteCapabilities[remoteServer].spreed = response.data.ocs.data
	remoteCapabilities[remoteServer].hash = joinRoomResponse.headers['x-nextcloud-talk-proxy-hash']
	remoteCapabilities[remoteServer].tokens.append(token) //maybe check existence before
```

This way, if a new conversation joined from already know remote server, it will just update it and not overwrite known tokens list. It will later be stored in browser storage and used in initialization, instead of going through the whole list of conversations (local and remote)

Reply:
New conversation won't go that far, if `hash` is equal to `x-nextcloud-talk-proxy-hash`.
I decided to drop storing tokens list in each remoteCaps in favor of tokenMap